crates/services/p2p/src/peer_manager.rs

```diff
@@ -199,13 +198,13 @@ impl NetworkBehaviour for PeerManagerBehaviour {
                         connection_established,
                     ));
 
-                let addresses = self.addresses_of_peer(&peer_id);
-                self.insert_peer_addresses(&peer_id, addresses);
-
                 if other_established == 0 {
                     // this is the first connection to a given Peer
                     self.peer_manager.handle_initial_connection(peer_id);
                 }
+
+                let addresses = self.addresses_of_peer(&peer_id);
```

Comment:
First, we need to insert the peer into the table in the `handle_initial_connection`, and after we can add peer addresses=)